packages/navigation/src/UIGridView/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { ListRenderItem, View, StyleSheet } from 'react-native';
+import { FlatList } from '..';
+import { UIConstant } from '../constants';
+import type { GridProps } from './types';
+
+export function UIGridView<T>({
+    data,
+    renderItem: renderItemProp,
+    containerStyle,
+    itemHeight,
+    testID,
+}: GridProps<T>) {
+    const [squareSize, setSquareSize] = React.useState({ width: 0, height: 0 });
+
+    const renderItem: ListRenderItem<T> = React.useCallback(
+        ({ item, index, separators }) => {
+            return (
+                <View style={[styles.item, squareSize]}>
+                    {renderItemProp?.({ item, index, separators })}
+                </View>
+            );
+        },
+        [renderItemProp, squareSize],
+    );
+
+    const onLayout = React.useCallback(
+        (event: any) => {
+            const { width } = event.nativeEvent.layout;
+            const squareDimension = width / UIConstant.grid.numColumns - 24;
+
+            setSquareSize({
+                width: squareDimension,
+                height: itemHeight ?? squareDimension,
+            });
+        },
+        [itemHeight],
+    );
+
+    return (
+        <View testID={testID} style={[styles.container, containerStyle]} onLayout={onLayout}>
```

Comment:
I'm not sure that we should wrap it with View, first `padding` is going to break scrollingIndicator from a right edge, second `flex: 1` makes impossible to have intrinsic size. I would suggest to remove it and use plain FlatList